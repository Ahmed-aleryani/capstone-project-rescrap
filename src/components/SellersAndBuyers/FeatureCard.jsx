import React from 'react';
import Title from '../Title/Title';
import Description from '../Description/Description';

const FeatureCard = ({imgPath, imgDesc, featHeader, featDesc}) => {
    return ( <div className="w-64 text-center">
        <img className="h-24 w-24 inline-block bg-green-400" src={imgPath} alt={imgDesc}/>
        <Title classes=" my-3" text={featHeader}/>
        <Description classes="" text={featDesc}/>
    </div> );
}
 
export default FeatureCard;