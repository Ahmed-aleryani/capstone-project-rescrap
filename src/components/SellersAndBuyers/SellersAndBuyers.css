.sellersBuyers {
  background-image: url(../../images/bg6.jpg);
  /* background-position: center; */
  background-size: cover;
  /* background-repeat: no-repeat; */
}
