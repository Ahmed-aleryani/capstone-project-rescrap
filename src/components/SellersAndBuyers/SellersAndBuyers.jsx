import React, { useState } from 'react';
import Title from '../Title/Title';
import SubTitle from '../SubTitle/SubTitle';
import FeatureCard from './FeatureCard';
import './SellersAndBuyers.css';
import seller1 from '../../images/seller1.jpg';
import seller2 from '../../images/seller2.jpg';
import seller3 from '../../images/seller3.jpg';
import buyer1 from '../../images/buyer1.jpg';
import buyer2 from '../../images/buyer2.jpg';
import buyer3 from '../../images/buyer3.jpg';

const SellersAndBuyers = () => {

    const sellersAndBuyersClasses= "grid sm\:row-gap-10 sm\:grid-cols-1 md\:grid-cols-2 lg\:grid-cols-3 xl\:grid-cols-3  col-gap-32 mx-32 my-20"
    const subHeadersClasses=" text-green-200 inline-block m-4";

    let [sellerHeader,setSellerHeader]=useState(subHeadersClasses+" underline");
    let [buyerHeader,setBuyerHeader]=useState(subHeadersClasses);

    let [sellerClasses,setSellerClasses]=useState(sellersAndBuyersClasses);
    let [buyerClasses,setBuyerClasses]=useState(sellersAndBuyersClasses+" hidden");

    const sellerHandeler=()=>{
        setSellerClasses(sellersAndBuyersClasses+" visible");
        setBuyerClasses(sellersAndBuyersClasses+" hidden");
        setSellerHeader(subHeadersClasses+" underline");
        setBuyerHeader(subHeadersClasses);
    }
    const buyerHandeler=()=>{
        setBuyerClasses(sellersAndBuyersClasses+" visible");
        setSellerClasses(sellersAndBuyersClasses+" hidden");
        setSellerHeader(subHeadersClasses);
        setBuyerHeader(subHeadersClasses+" underline");
    }

    return ( <div className="sellersBuyers">
        <div className="text-center pt-24">
        <Title classes=" text-green-200" text="HOW IT WORKS"/>
        <a  onClick={sellerHandeler} className="cursor-pointer"><SubTitle classes={sellerHeader} text="Sellers"/></a>
        <a  onClick={buyerHandeler} className="cursor-pointer"><SubTitle classes={buyerHeader} text="Buyers"/></a>
        </div>
        <div className={sellerClasses}>
            <FeatureCard imgPath={seller1} imgDesc="Post your offer" featHeader="Post your offer" featDesc="Join the marketplace and post products you want to sell in less than a minute."/>
            <FeatureCard imgPath={seller2} imgDesc="Get the best price" featHeader="Get the best price" featDesc="Negotiate with buyers in real-time and get the best price for your products."/>
            <FeatureCard imgPath={seller3} imgDesc="Sell to customers" featHeader="Sell to customers" featDesc="Sell products directly to customers in your country and around the world."/>
        </div>
        <div className={buyerClasses}>
            <FeatureCard imgPath={buyer1} imgDesc="Find what you’re looking for" featHeader="Find what you’re looking for" featDesc="Browse through thousands of offers posted in the marketplace and find what you’re looking for."/>
            <FeatureCard imgPath={buyer2} imgDesc="Negotiate the best deal" featHeader="Negotiate the best deal" featDesc="Negotiate with multiple sellers in real-time and make the best deal."/>
            <FeatureCard imgPath={buyer3} imgDesc="Buy directly from suppliers" featHeader="Buy directly from suppliers" featDesc="Buy directly from suppliers of quality material in your country or internationally."/>
        </div>
        </div> );
}
 
export default SellersAndBuyers;