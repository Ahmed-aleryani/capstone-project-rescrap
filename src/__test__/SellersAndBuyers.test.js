import React from 'react';
import renderer from 'react-test-renderer';
import SellersAndBuyers from '../components/SellersAndBuyers/SellersAndBuyers';


it("matches the snapshot",()=>{
    const tree = renderer.create(<SellersAndBuyers/>).toJSON();
    expect(tree).toMatchSnapshot();
})