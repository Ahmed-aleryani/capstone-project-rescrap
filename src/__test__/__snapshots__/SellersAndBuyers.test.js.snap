// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="sellersBuyers"
>
  <div
    className="text-center pt-24"
  >
    <h1
      className="text-3xl normal text-green-200"
    >
      HOW IT WORKS
    </h1>
    <a
      className="cursor-pointer"
      onClick={[Function]}
    >
      <h2
        className="text-2xl thin text-green-200 inline-block m-4 underline"
      >
        Sellers
      </h2>
    </a>
    <a
      className="cursor-pointer"
      onClick={[Function]}
    >
      <h2
        className="text-2xl thin text-green-200 inline-block m-4"
      >
        Buyers
      </h2>
    </a>
  </div>
  <div
    className="grid sm:row-gap-10 sm:grid-cols-1 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-3  col-gap-32 mx-32 my-20"
  >
    <div
      className="w-64 sm:pb-16 text-center undefined"
    >
      <img
        alt="Post your offer"
        className="h-24 w-24 inline-block bg-green-400"
        src="seller1.jpg"
      />
      <h2
        className="text-2xl thin my-3"
      >
        Post your offer
      </h2>
      <p
        className="text-base font-normal "
      >
        Join the marketplace and post products you want to sell in less than a minute.
      </p>
    </div>
    <div
      className="w-64 sm:pb-16 text-center undefined"
    >
      <img
        alt="Get the best price"
        className="h-24 w-24 inline-block bg-green-400"
        src="seller2.jpg"
      />
      <h2
        className="text-2xl thin my-3"
      >
        Get the best price
      </h2>
      <p
        className="text-base font-normal "
      >
        Negotiate with buyers in real-time and get the best price for your products.
      </p>
    </div>
    <div
      className="w-64 sm:pb-16 text-center undefined"
    >
      <img
        alt="Sell to customers"
        className="h-24 w-24 inline-block bg-green-400"
        src="seller3.jpg"
      />
      <h2
        className="text-2xl thin my-3"
      >
        Sell to customers
      </h2>
      <p
        className="text-base font-normal "
      >
        Sell products directly to customers in your country and around the world.
      </p>
    </div>
  </div>
  <div
    className="grid sm:row-gap-10 sm:grid-cols-1 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-3  col-gap-32 mx-32 my-20 hidden"
  >
    <div
      className="w-64 sm:pb-16 text-center undefined"
    >
      <img
        alt="Find what you’re looking for"
        className="h-24 w-24 inline-block bg-green-400"
        src="buyer1.jpg"
      />
      <h2
        className="text-2xl thin my-3"
      >
        Find what you’re looking for
      </h2>
      <p
        className="text-base font-normal "
      >
        Browse through thousands of offers posted in the marketplace and find what you’re looking for.
      </p>
    </div>
    <div
      className="w-64 sm:pb-16 text-center undefined"
    >
      <img
        alt="Negotiate the best deal"
        className="h-24 w-24 inline-block bg-green-400"
        src="buyer2.jpg"
      />
      <h2
        className="text-2xl thin my-3"
      >
        Negotiate the best deal
      </h2>
      <p
        className="text-base font-normal "
      >
        Negotiate with multiple sellers in real-time and make the best deal.
      </p>
    </div>
    <div
      className="w-64 sm:pb-16 text-center undefined"
    >
      <img
        alt="Buy directly from suppliers"
        className="h-24 w-24 inline-block bg-green-400"
        src="buyer3.jpg"
      />
      <h2
        className="text-2xl thin my-3"
      >
        Buy directly from suppliers
      </h2>
      <p
        className="text-base font-normal "
      >
        Buy directly from suppliers of quality material in your country or internationally.
      </p>
    </div>
  </div>
</div>
`;
